### ISSUE TITLE
ExpansionTileThemeData shape not working

### ISSUE THREAD

**Issue by a1573595** [flutter/flutter#134488]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Run sample code.

### Expected results

ExpansionTile will change the border when ExpansionTileThemeData sets a different shape.

### Actual results

Not work, I need to set the shape on every ExpansionTile.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        expansionTileTheme: const ExpansionTileThemeData(
          /// Not working
          shape: Border(),
        ),
      ),
      home: HomePage(),
    );
  }
}

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: const Column(
        children: [
          SizedBox(
            height: 32,
          ),

          /// ExpansionTileThemeData border not working
          ExpansionTile(
            initiallyExpanded: true,
            title: Text('ExpansionTile'),
            children: [
              SizedBox(
                height: 32,
              )
            ],
          ),
          SizedBox(
            height: 32,
          ),
          ExpansionTile(
            initiallyExpanded: true,

            /// It is work
            shape: Border(),
            title: Text('ExpansionTile set shape'),
            children: [
              SizedBox(
                height: 32,
              )
            ],
          ),
        ],
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

<img width="344" alt="截圖 2023-09-12 上午9 05 40" src="https://github.com/flutter/flutter/assets/25738593/4079ff77-0598-42f8-a063-4a7d6b7fc92e">

</details>

### Logs

<details><summary>Logs</summary>

```console
Performing hot reload...
Syncing files to device sdk gphone64 arm64...
Reloaded 1 of 743 libraries in 318ms (compile: 73 ms, reload: 93 ms, reassemble: 77 ms).
D/EGL_emulation(31369): app_time_stats: avg=3512.53ms min=15.12ms max=20992.66ms count=6
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.3, on macOS 13.5.2 22G91 darwin-arm64, locale zh-Hant-TW)
    • Flutter version 3.13.3 on channel stable at /Library/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2524052335 (5 days ago), 2023-09-06 14:32:31 -0700
    • Engine revision b8d35810e9
    • Dart version 3.1.1
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/chien/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • ANDROID_SDK_ROOT = /Users/chien/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.82.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.72.0

[✓] Connected device (4 available)
    • ASUS Z01RD (mobile)         • J9AZB760H2767RP • android-arm64  • Android 10 (API 29)
    • sdk gphone64 arm64 (mobile) • emulator-5554   • android-arm64  • Android 13 (API 33) (emulator)
    • macOS (desktop)             • macos           • darwin-arm64   • macOS 13.5.2 22G91 darwin-arm64
    • Chrome (web)                • chrome          • web-javascript • Google Chrome 116.0.5845.179

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by dam-ease** [comment]
Hi @a1573595. Thanks for filing this. I can reproduce this on the latest `stable` channel but can’t reproduce on `master`. I see there is a similar issue https://github.com/flutter/flutter/issues/129785  that has been fixed. Are you able to check this on the master channel?

**Comment by a1573595** [comment]
> Hi @a1573595. Thanks for filing this. I can reproduce this on the latest `stable` channel but can’t reproduce on `master`. I see there is a similar issue #129785 that has been fixed. Are you able to check this on the master channel?
> 
> stable, master flutter doctor -v
> ```
> [✓] Flutter (Channel stable, 3.13.3, on macOS 13.0 22A380 darwin-arm64, locale
>     en-NG)
>     • Flutter version 3.13.3 on channel stable at
>       /Users/damilolaalimi/sdks/flutter
>     • Upstream repository https://github.com/flutter/flutter.git
>     • Framework revision 2524052335 (5 days ago), 2023-09-06 14:32:31 -0700
>     • Engine revision b8d35810e9
>     • Dart version 3.1.1
>     • DevTools version 2.25.0
> 
> [✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
>     • Android SDK at /Users/damilolaalimi/Library/Android/sdk
>     • Platform android-34, build-tools 34.0.0
>     • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
>     • Java binary at: /Applications/Android
>       Studio.app/Contents/jbr/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build
>       17.0.6+0-17.0.6b802.4-9586694)
>     • All Android licenses accepted.
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Build 14E300c
>     • CocoaPods version 1.12.1
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [✓] Android Studio (version 2022.2)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
>     • Java version OpenJDK Runtime Environment (build
>       17.0.6+0-17.0.6b802.4-9586694)
> 
> [✓] VS Code (version 1.81.1)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension version 3.50.0
> 
> [✓] Connected device (6 available)
>     • sdk gphone64 arm64 (mobile) • emulator-5554                        •
>       android-arm64  • Android 14 (API 34) (emulator)
>     • iPhone 13 Pro (mobile)      • 14B6FEE4-2F1C-401F-8804-A5115D7A6DA2 • ios
>       • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
>     • iPhone 14 (mobile)          • 1A122DE2-0CAB-4C3E-A395-691BF27D626F • ios
>       • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
>     • iPhone 14 Pro (mobile)      • 6548B0D1-7674-4306-9852-198C465F1104 • ios
>       • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
>     • macOS (desktop)             • macos                                •
>       darwin-arm64   • macOS 13.0 22A380 darwin-arm64
>     • Chrome (web)                • chrome                               •
>       web-javascript • Google Chrome 116.0.5845.179
> 
> [✓] Network resources
>     • All expected network resources are available.
> 
> • No issues found!
> ```
> 
> ```
> [✓] Flutter (Channel master, 3.14.0-14.0.pre.211, on macOS 13.0 22A380 darwin-arm64, locale en-NG)
>     • Flutter version 3.14.0-14.0.pre.211 on channel master at /Users/damilolaalimi/fvm/versions/master
>     • Upstream repository https://github.com/flutter/flutter.git
>     • Framework revision 69f6e7d15d (16 hours ago), 2023-09-11 00:50:38 -0400
>     • Engine revision 7e41b064eb
>     • Dart version 3.2.0 (build 3.2.0-134.0.dev)
>     • DevTools version 2.27.0
> 
> [✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
>     • Android SDK at /Users/damilolaalimi/Library/Android/sdk
>     • Platform android-34, build-tools 34.0.0
>     • ANDROID_HOME = /Users/damilolaalimi/Library/Android/sdk
>     • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
>     • All Android licenses accepted.
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Build 14E300c
>     • CocoaPods version 1.12.1
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [✓] Android Studio (version 2022.2)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
>     • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
> 
> [✓] VS Code (version 1.81.1)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension version 3.50.0
> 
> [✓] Connected device (6 available)
>     • sdk gphone64 arm64 (mobile) • emulator-5554                        • android-arm64  • Android 14 (API 34) (emulator)
>     • iPhone 13 Pro (mobile)      • 14B6FEE4-2F1C-401F-8804-A5115D7A6DA2 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
>     • iPhone 14 (mobile)          • 1A122DE2-0CAB-4C3E-A395-691BF27D626F • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
>     • iPhone 14 Pro (mobile)      • 6548B0D1-7674-4306-9852-198C465F1104 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
>     • macOS (desktop)             • macos                                • darwin-arm64   • macOS 13.0 22A380 darwin-arm64
>     • Chrome (web)                • chrome                               • web-javascript • Google Chrome 116.0.5845.187
> 
> [✓] Network resources
>     • All expected network resources are available.
> 
> • No issues found!
> ```

It works on the master channel, but #129785 was fixed on 2023/7/12, Flutter SDK 3.13 was released on 2023/8/17, the timeline doesn't let me determine if the bug has been fixed or reappeared, switch to the master channel is a way of judging?

**Comment by dam-ease** [comment]
Thanks for confirming this @a1573595. You can read about it here: https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#will-a-particular-bug-fix-be-provided-in-a-hotfix-release (and whole that page for more info)
Thank you!
